### PR TITLE
Add access controls

### PR DESCRIPTION
**Pull request by SlimaneAmar**

### CODE REVIEW

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 1]
```diff
@@ -195,43 +216,79 @@ private void emitDirectoryChanged(UUID directoryUuid, String userId, String erro
             .then();
     }
 
+    //
```
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:219]: empty comment

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 2]
```diff
@@ -195,43 +216,79 @@ private void emitDirectoryChanged(UUID directoryUuid, String userId, String erro
...
+        if (element.getType().equals(DIRECTORY)) {
+            return element.getOwner().equals(userId) && // Only owner
+                (!forDeletion || getDirectoryElementsStream(element.getElementUuid())
+                    .filter(e -> e.getType().equals(DIRECTORY))
+                    .allMatch(e -> isElementUpdatable(e, userId, true))
+                );
+        } else {
+            return getParentElement(element.getElementUuid()).isAllowed(userId);
+        }
+    }
```
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:230]: We agreed that if it's public everyone can do whatever they want with it including deleting it

**Review by AbdelHedhili — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/directory/server/dto/ElementAttributes.java [hunk 3]
```diff
@@ -40,6 +41,10 @@
 
     private String description;
 
+    public boolean isAllowed(@NonNull String userId) {
+        return owner.equals(userId) || !accessRights.isPrivate();
```
Comment (AbdelHedhili) [inline src/main/java/org/gridsuite/directory/server/dto/ElementAttributes.java:45]: it's scary to have this method here. It works fine if we call it on a directory element but it doesn't in any other case as we want to know the accessRights of the parent

**Review by AbdelHedhili — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java (hunk 1 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/directory/server/DirectoryService.java:219]: Removed

File: src/main/java/org/gridsuite/directory/server/dto/ElementAttributes.java (hunk 3 above)
Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/directory/server/dto/ElementAttributes.java:45]: I agree but we don't have a dto dedicated to a folder (as for entities)

Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/directory/server/dto/ElementAttributes.java:50]: Control added

**Review by AbdelHedhili — APPROVED** [review]

**Review by AbdelHedhili — APPROVED** [review]

File: src/main/java/org/gridsuite/directory/server/repository/DirectoryElementRepository.java [hunk 4]
```diff
@@ -22,34 +21,25 @@
...
 
-    @Query("SELECT d FROM DirectoryElementEntity d  WHERE d.parentId = :parentId AND (d.isPrivate='false' or d.owner=:userId)")
-    List<DirectoryElementEntity> findDirectoryContentByUserId(UUID parentId, String userId);
+    List<DirectoryElementEntity> findAllByParentId(UUID parentId);
 
     @Query("SELECT d FROM DirectoryElementEntity d  WHERE d.parentId IS NULL AND d.type = 'DIRECTORY' AND (d.isPrivate='false' or d.owner=:owner)")
     List<DirectoryElementEntity> findRootDirectoriesByUserId(String owner);
 
+    @Query("SELECT d FROM DirectoryElementEntity d  WHERE d.parentId IS NULL AND d.type = 'DIRECTORY' AND d.name=:name")
+    List<DirectoryElementEntity> findRootDirectoriesByName(String name);
```
Comment (JacquesBorsenberger) [inline src/main/java/org/gridsuite/directory/server/repository/DirectoryElementRepository.java:30]: we can have multiple root with the same name ?

Comment (SlimaneAmar) [reply src/main/java/org/gridsuite/directory/server/repository/DirectoryElementRepository.java:30]: Yes  it's planned
See discussions in Slack

**Pull request merged**